websocketOutportDriver/sender/websocketClientsHolder.go

```diff
@@ -0,0 +1,55 @@
+package sender
+
+import "sync"
+
+type websocketClientsHolder struct {
+	clients map[string]*webSocketClient
+	mut     sync.RWMutex
+}
+
+// NewWebsocketClientsHolder will return a new instance of websocketClientsHolder
+func NewWebsocketClientsHolder() *websocketClientsHolder {
+	return &websocketClientsHolder{
+		clients: make(map[string]*webSocketClient),
+	}
+}
+
+// AddClient will add the provided client to the internal members
+func (wch *websocketClientsHolder) AddClient(client *webSocketClient) {
+	wch.mut.Lock()
+	wch.clients[client.remoteAddr] = client
+	wch.mut.Unlock()
+}
+
+// GetClient will return a client based on the provided remote address
+func (wch *websocketClientsHolder) GetClient(remoteAddr string) (*webSocketClient, bool) {
```

Comment:
The way the code is written right now, the values aren't modified by the caller. I'll leave it as it is (removed the GetClient function, tho)

Reply:
Right, because in the current code it is used correctly.
But if a new dev will work on these files, he could wrongly use the functions.
Perhaps just add a comment/TODO to raise awareness of the problem

Reply:
added a todo: `// TODO: use interfaces instead of direct instances + analyze returning pointers vs values on exported functions`